internal/http/compress_test.go

```diff
@@ -0,0 +1,96 @@
+// +build unit
+
+package http
+
+import (
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	mock "github.com/newrelic/newrelic-client-go/pkg/testhelpers"
+)
+
+var testCompressionCases = []struct {
+	compress bool
```

Comment:
If compress is false, we expect the data to NOT be compressed by the compressor, and instead compare data == data after running through the compressor.